packages/ra-ui-materialui/src/list/SimpleList.tsx

```diff
@@ -95,15 +96,20 @@ const SimpleList: FC<SimpleListProps> = props => {
     return (
         total > 0 && (
             <List className={className} {...sanitizeListRestProps(rest)}>
-                {ids.map(id => (
+                {ids.map((id, rowIndex) => (
                     <LinkOrNot
                         linkType={linkType}
                         basePath={basePath}
                         id={id}
                         key={id}
                         record={data[id]}
                     >
-                        <ListItem button={!!linkType as any}>
+                        <ListItem
+                            button={!!linkType as any}
+                            style={
+                                rowStyle ? rowStyle(data[id], rowIndex) : null
```

Comment:
It seems you didn't replace `null` with `undefined` as requested by @jdemangeon. Can you please fix that?

Reply:
I have but accidentially reverted this one also. Brought it back again, sorry.